README
Dist::Zilla::App::Command::dumpwith

This command, like its sibling "dumpphases", exists to help make
understanding what is going on in "Dist::Zilla" a little easier.

At least, having this command means debugging certain kinds of problems
is more obvious.

If you want to see all plugins that are adding files to your dist?

    dzil dumpwith -- -FileGatherer

Though, of course, this requires some knowledge of what roles are
applicable.

If you want to turn colors off, use "Term::ANSIcolor"'s environment
variable "ANSI_COLORS_DISABLED". E.g.,

    ANSI_COLORS_DISABLED=1 dzil dumpphases

Alternatively, specify a color-free theme:

    dzil dumpwith --color-theme=basic::plain -- -VersionProvider

INSTALLATION

This is a Perl module distribution. It should be installed with whichever
tool you use to manage your installation of Perl, e.g. any of

  cpanm .
  cpan  .
  cpanp -i .

Consult http://www.cpan.org/modules/INSTALL.html for further instruction.
Should you wish to install this module manually, the procedure is

  perl Makefile.PL
  make
  make test
  make install

COPYRIGHT AND LICENSE

This software is copyright (c) 2016 by Kent Fredric
<kentfredric@gmail.com>.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


Changes
Release history for Dist-Zilla-App-Command-dumpwith

0.003003 2017-03-07T07:46:21Z 848a78c

0.003002 2017-03-07T07:45:06Z 59589aa
 [Bugfix]
 - Fix test failures with -Ddefault_inc_excludes_dot

 [Dependencies::Stats]
 - Dependencies changed since 0.003001, see misc/*.deps* for details
 - develop: +1 ↑1 (suggests: ↑2)
 - test: ↓1

0.003001 2016-06-06T21:15:46Z 78c1547
 - Documentation: Update screenshots to reflect new notation.
 - Documentation: Move KNOWN ISSUES further down.

0.003000 2016-05-28T16:13:51Z 2b26a07
 - Add support for "--" escaping to isolate the role names to avoid complications introduced by '-V' becomming a thing
   on Dzil 6.0.
 - Tests and documentation also updated to prefer that notation where it could be ambiguous.
 - Closes #1, thanks to Slaven Rezić and Karen Etheridge for bug feedback.

 [Dependencies::Stats]
 - Dependencies changed since 0.002012, see misc/*.deps* for details
 - develop: +5 ↑1 -1 (suggests: ↑2)
 - test: +2 -1

 [Tests]
 - Be less fragile and stop using my dztest thing. Its overcomplicated and I don't need it anymore.

0.002012 2015-04-15T00:35:51Z 59708df
 [Dependencies]
 - Make sure tests have a new enough version of dztest for app testing.

 [Dependencies::Stats]
 - Dependencies changed since 0.002011, see misc/*.deps* for details
 - test: ↑1

0.002011 2015-04-13T02:42:56Z e9deaed
 [Documentation]
 - Redirect URIs to different repo.

 [Packaging]
 - Removed some large unneeded files

0.002010 2015-04-13T01:38:43Z 3258308
 [Dependencies::Stats]
 - Dependencies changed since 0.002003, see misc/*.deps* for details
 - develop: +3 ↑3 -1 (recommends: ↓1, suggests: ↑2)
 - runtime: +2 (recommends: +1)
 - test: +3 (recommends: ↓1)

 [Internals]
 - Theme loading and construction simplified
 - Theme defaulting uses less useless code.
 - Color themes validated earlier
 - Load Dzil lazier

 [Tests]
 - Added tests covering noop modes, explicit named and prefixed roles, and color theme support and validation.

0.002003 2014-10-31T15:02:11Z 567b541
 [Dependencies::Stats]
 - Dependencies changed since 0.002002, see misc/*.deps* for details
 - configure: +1 (recommends: ↑1)
 - develop: +6 ↑2 -1 (recommends: +1, suggests: ↑2)
 - runtime: ↓1 -2
 - test: +1 (recommends: ↑2)

 [Internals]
 - remove utf8
 - now supports 5.006
 - remove unneeded autoboxing
 - defer all modules possible

 [Metadata]
 - author = kentnl@cpan.org

0.002002 2014-08-15T21:14:38Z
 [00 Trivial]
 - CPANDAY!
 - no code changes.
 - tests augmented.
 - whitespace adjusted.

 [Dependencies::Stats]
 - Dependencies changed since 0.002001, see misc/*.deps* for details
 - develop: +1 (suggests: ↑1)
 - test: -3

 [Misc]
 - Improve source side POD to reduce build side whitespace.

 [Tests]
 - update Test::ReportPrereqs

0.002001 2014-07-31T10:07:33Z
 [Bugfix]
 - Fix doc link to screenshot

0.002000 2014-07-31T09:56:35Z
 [00 Major]
 - Changes that may affect hand-operated downstream.

 [01 Toolkit]
 - Switched To ExtUtils::MakeMaker
 - Toolkit dependenecies for "latest" only "recommeneded" instead of required.
 - https://questhub.io/realm/perl/quest/5388cc12bbd0be5b2500011f

 [Dependencies::Stats]
 - Dependencies changed since 0.001000, see misc/*.deps* for details
 - build: -1
 - configure: +1 -1 (recommends: +1)
 - develop: +9 -11 (suggests: +2 -1)
 - runtime: +1 ↑1
 - test: +5 ↓1 (recommends: +3)

0.001000 2013-11-11T13:48:41Z
 - First version.


Build.PL

# This file was automatically generated by Dist::Zilla::Plugin::ModuleBuild v5.020.
use strict;
use warnings;

use Module::Build 0.28;


my %module_build_args = (
  "build_requires" => {
    "Module::Build" => "0.4206"
  },
  "configure_requires" => {
    "Module::Build" => "0.4206"
  },
  "dist_abstract" => "Dump all plugins that 'do' a certain role",
  "dist_author" => [
    "Kent Fredric <kentfredric\@gmail.com>"
  ],
  "dist_name" => "Dist-Zilla-App-Command-dumpwith",
  "dist_version" => "0.001001",
  "license" => "perl",
  "module_name" => "Dist::Zilla::App::Command::dumpwith",
  "recommends" => {},
  "recursive_test_files" => 1,
  "requires" => {
    "Carp" => 0,
    "Dist::Zilla::App" => 0,
    "Dist::Zilla::dumpphases::Theme::basic::blue" => 0,
    "Module::Runtime" => 0,
    "Moose::Autobox" => 0,
    "Scalar::Util" => 0,
    "Try::Tiny" => 0,
    "perl" => "5.006",
    "strict" => 0,
    "warnings" => 0
  },
  "script_files" => [],
  "test_requires" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "File::Spec::Functions" => 0,
    "List::Util" => 0,
    "Scalar::Util" => 0,
    "Test::More" => "1.001003"
  }
);


my %fallback_build_requires = (
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "File::Spec::Functions" => 0,
  "List::Util" => 0,
  "Module::Build" => "0.4206",
  "Scalar::Util" => 0,
  "Test::More" => "1.001003"
);


unless ( eval { Module::Build->VERSION(0.4004) } ) {
  delete $module_build_args{test_requires};
  $module_build_args{build_requires} = \%fallback_build_requires;
}

my $build = Module::Build->new(%module_build_args);

$build->create_build_script;


maint/perlcritic.rc.gen.pl
#!/usr/bin/env perl
## no critic (Modules::RequireVersionVar)

# ABSTRACT: Write an INI file from a bundle

use 5.008;    # utf8
use strict;
use warnings;
use utf8;

our $VERSION = 0.001;

use Carp qw( croak carp );
use Perl::Critic::ProfileCompiler::Util qw( create_bundle );
use Path::Tiny qw(path);

## no critic (ErrorHandling::RequireUseOfExceptions)
my $bundle = create_bundle('Example::Author::KENTNL');
$bundle->configure;

my @stopwords = (
  qw(
    screenshot
    )
);
for my $wordlist (@stopwords) {
  $bundle->add_or_append_policy_field( 'Documentation::PodSpelling' => ( 'stop_words' => $wordlist ) );
}

#$bundle->add_or_append_policy_field(
#  'Subroutines::ProhibitCallsToUndeclaredSubs' => ( 'exempt_subs' => 'String::Formatter::str_rf' ), );

$bundle->remove_policy('ErrorHandling::RequireUseOfExceptions');

#$bundle->remove_policy('CodeLayout::RequireUseUTF8');

#$bundle->remove_policy('ErrorHandling::RequireCarping');
$bundle->remove_policy('NamingConventions::Capitalization');

my $inf = $bundle->actionlist->get_inflated;

my $config = $inf->apply_config;

{
  my $rcfile = path('./perlcritic.rc')->openw_utf8;
  $rcfile->print( $config->as_ini, "\n" );
  close $rcfile or croak 'Something fubared closing perlcritic.rc';
}
my $deps = $inf->own_deps;
{
  my $target = path('./misc');
  $target->mkpath if not $target->is_dir;

  my $depsfile = $target->child('perlcritic.deps')->openw_utf8;
  for my $key ( sort keys %{$deps} ) {
    $depsfile->printf( "%s~%s\n", $key, $deps->{$key} );
    *STDERR->printf( "%s => %s\n", $key, $deps->{$key} );
  }
  close $depsfile or carp 'Something fubared closing perlcritic.deps';
}



Makefile.PL
# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v6.009.
use strict;
use warnings;

use 5.006;

use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "Dump all plugins that 'do' a certain role",
  "AUTHOR" => "Kent Fredric <kentnl\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "Dist-Zilla-App-Command-dumpwith",
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.006",
  "NAME" => "Dist::Zilla::App::Command::dumpwith",
  "PREREQ_PM" => {
    "Carp" => 0,
    "Dist::Zilla::App" => 0,
    "Dist::Zilla::dumpphases::Theme::basic::blue" => 0,
    "Module::Runtime" => 0,
    "Path::ScanINC" => 0,
    "Path::Tiny" => 0,
    "Scalar::Util" => 0,
    "Try::Tiny" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "Dist::Zilla::App::Tester" => 0,
    "Dist::Zilla::Plugin::AutoVersion" => 0,
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "Test::DZil" => 0,
    "Test::More" => 0,
    "Test::TempDir::Tiny" => 0
  },
  "VERSION" => "0.003003",
  "test" => {
    "TESTS" => "t/*.t t/00-compile/*.t"
  }
);


my %FallbackPrereqs = (
  "Carp" => 0,
  "Dist::Zilla::App" => 0,
  "Dist::Zilla::App::Tester" => 0,
  "Dist::Zilla::Plugin::AutoVersion" => 0,
  "Dist::Zilla::dumpphases::Theme::basic::blue" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "Module::Runtime" => 0,
  "Path::ScanINC" => 0,
  "Path::Tiny" => 0,
  "Scalar::Util" => 0,
  "Test::DZil" => 0,
  "Test::More" => 0,
  "Test::TempDir::Tiny" => 0,
  "Try::Tiny" => 0,
  "strict" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);
